### PR TITLE
Feat/#1801 access facet feedback

### PR DESCRIPTION
**Pull request by ciaranschutte**
**Description of changes**

- Access facet should be very first facet, pulled facet out of out loop to show before search bar
- Names in querybar should be readable, added more to `displayValue` function
- order of options now fixed

**Type of Change**

- [ ] Bug
- [ ] Styling
- [ ] New Feature

**Checklist before requesting review:**

- [x] Matches design:

  - component sizes, spacing, and styles
  - font size, weight, colour
  - spelling has been double checked

- [ ] New uikit components have Storybook stories
- [ ] Feature is minimally responsive.
- [ ] Add copyrights to new files
- [x] Manual testing of UI feature.
- [ ] Selenium test is completed and passing.

### CODE REVIEW

File: components/pages/file-repository/FacetPanel/index.tsx [hunk 1]
```diff
@@ -275,6 +276,22 @@ export default () => {
...
+  const orderReleaseStage = (options: FilterOption[]) => {
+    const order = ['OWN_PROGRAM', 'FULL_PROGRAMS', 'ASSOCIATE_PROGRAMS', 'PUBLIC_QUEUE', 'PUBLIC'];
+    return order.map((order) => options.find((option) => option.key === order)).filter(Boolean);
+  };
+
+  // function to manipulate options values eg. to preserve order
+  const prepareOptions = (
+    options: FilterOption[],
+    prepare: (options: FilterOption[]) => FilterOption[],
+  ) => prepare(options);
```
Comment (joneubank) [inline components/pages/file-repository/FacetPanel/index.tsx:293]: Why make a function that includes the function you want to use as an argument and does nothing else? Especially if the default is a "do nothing" function? Function.

I don't understand the value here, it just seems like an extra layer of complication.

File: components/pages/file-repository/FacetPanel/index.tsx [hunk 2]
```diff
@@ -416,6 +434,12 @@ export default () => {
     },
   });
 
+  // pull out release stage facet to show before search bar
+  const releaseStageDetails = presetFacets.filter(
+    (f) => f.facetPath === FileFacetPath.release_stage,
+  )[0];
```
Comment (joneubank) [inline components/pages/file-repository/FacetPanel/index.tsx:440]: This looks like what you want is the `.find` method.
```
const releaseStageDetails = presetFacets.find( f => f.facetPath === FileFacetPath.release_stage);
```

File: components/pages/file-repository/FacetPanel/index.tsx [hunk 3]
```diff
@@ -534,45 +574,35 @@ export default () => {
           </MenuItem>
         </FacetRow>
         {!loadingFieldDisplayNames &&
-          presetFacets.map((facetDetails) => {
-            const facetProps = commonFacetProps(facetDetails);
+          presetFacets
+            .filter((f) => f.facetPath !== FileFacetPath.release_stage)
```
Comment (joneubank) [inline components/pages/file-repository/FacetPanel/index.tsx:578]: Maybe add a comment here stating why we are filtering out this particular facet (its obvious if you read the above section, but if you just look here then this code seems odd).

**Review by joneubank — CHANGES_REQUESTED** [review]

File: components/pages/file-repository/utils/index.ts [hunk 4]
```diff
@@ -226,10 +226,20 @@ export const removeFilter: TRemoveFilter = (field, filters) => {
...
+export const toDisplayValue: (value: string, field?: string) => string = (value, field) => {
+  if (value === IS_MISSING) {
+    return 'No Data';
+  }
+
+  if (!!field && !!value) {
+    const displayName = getDisplayName(field, value);
+    return displayName ? displayName : value;
+  } else {
+    return value;
```
[the remark this answers is not in the document]
Comment (ciaranschutte) [reply components/pages/file-repository/utils/index.ts:238]: we will always have a `value` or our type check will fail
but your point stands regarding the optional `field`

changing so if we get just a value, and it doesn't match any special display condition, value is simply returned

**Review by joneubank — APPROVED** [review]

**Pull request merged**